Provide timeout for requests

yoshson opened the pull request:
The requests api does not provide a default timeout. In synchronous setups this might cause problems, if for some reason the chosen plausible server does not respond.

I added a timeout to all proxied requests, not sure about the default of 1 sec though.

yoshson committed:
provide default request timeout

imankulov commented:
Hey @yoshson, great catch. Thanks a lot for your contribution. I think 1 sec as the default timeout is a decent starting point.

I can see your PR contains a few unrelated changes and one revert commit. Can you please delete commits 9df5b25 and  e599222 so that I can merge a cleaner version of the PR? 

Also, can you please update the README file and document PLAUSIBLE_REQUEST_TIMEOUT? Thanks!

imankulov commented:
Upd: In the main branch I fixed GitHub actions and re-applied your infrastructure changes. Feel free to rebase your branch on top of the current main. Tests should pass this time.

yoshson committed:
Add PLAUSIBLE_REQUEST_TIMEOUT paragraph

yoshson commented:
Hi @imankulov,

added a README paragraph and cleaned up the git logs. Sorry, i'm pretty new to contributing.
Thanks a lot for your time and for maintaining this package! 🙏

imankulov commented:
Awesome contribution, @yoshson. Thanks a lot for it!
